Add code and PDFs for bulk figures

sjspielman opened the pull request:
Closes #187 
Closes #195 
Stacked on #202 (review #202 first!)

This PR adds code to make the bulk manuscript figures. I originally planned to split this up across PRs, but this turned out to not be _that_ much code, and given how I ended up combing panels, a single PR ended up making more sense to me. 

We have two new pdf panels for the main text (brain only) and supplement (wilms & osteo). I included `(N=##)` labels in the faceting to indicate the number of samples in each. 

Some spots that would be great for feedback include:
- The color of the `geom_smooth()` line in the scatterplots. This color isn't the most visible, but given the color scale present for binned points it's hard to find a color which will remain highly visible across the binned color scale. This seemed close among different options I tried!
- Do we like the `geom_text()` for the odds ratio values, or should we keep this as plain bars sans text?
- At one point I had considered adding text into the X-axis labels for the consensus cell types to indicate the size of each gene set. But, the plot seems like it's busy enough and that information is more "bonus" and not strictly needed for interpretation. Does anyone think it's important enough to include?

Noting that this is ready for review, but I'm going to hold off requesting review in case any changes to this figure are discussed during our pre-planning meeting shortly.

jaclyn-taroni commented on scripts/Fig6-FigS7_bulk-analysis.R:
```diff
@@ -27,10 +38,81 @@ geneset_files <- list.files(
+    labs(
+      x = "Consensus cell type", 
+      y = "Odds ratio", 
+      fill = "-Log10(P)"
```
Might as well be a bit more verbose. We seem to have the room!
```suggestion
      fill = "-Log10(adj. p-value)"
```

sjspielman replied:
done

jaclyn-taroni commented on scripts/Fig6-FigS7_bulk-analysis.R:
```diff
@@ -27,10 +38,81 @@ geneset_files <- list.files(
+    aes(x = pseudobulk, y = bulk) + 
+    geom_bin_2d(binwidth = 0.25) + 
+    scale_fill_viridis_c(limits = c(0, 800), oob = scales::squish, option = "inferno") +
+    geom_smooth(method = "lm", color = "paleturquoise", linewidth = 0.75) + 
```
It's a little bit pale for the white background. What about `deepskyblue` or `cyan2`?

sjspielman replied:
cyan2 ✅

jaclyn-taroni commented on scripts/Fig6-FigS7_bulk-analysis.R:
```diff
@@ -27,10 +38,81 @@ geneset_files <- list.files(
+        label = round(odds_ratio, 2), 
+        y = odds_ratio + 0.75
+      ), 
+      size = 2.5
```
Can we try 3 or perhaps 2.75?

sjspielman replied:
3 looks good!

jaclyn-taroni commented on scripts/Fig6-FigS7_bulk-analysis.R:
```diff
@@ -94,6 +176,93 @@ bulk_table <- bulk_table |>
+
+
+# SI panel:
+odds_panel_si <- odds_df |>
+  dplyr::filter(scpca_project_id %in% si_projects) |>
+  plot_odds_ratios()
```
We have a little bit of an issue with the spacing of the `mononuclear phagocyte` and `dendritic cell labels` for `SCPCP000017`. I wonder if a slightly different angle just for the supplemental figure is the solution?

sjspielman replied:
I made the angle 45 all around, and I think it looks better in both supp and main panels.

jaclyn-taroni reviewed (commented):
This looks mostly good, and I think it's close, but I had a couple ideas for some small tweaks.

sjspielman commented:
Figures have been updated in response to feedback! I also updated the script name (this is S8, not S7!), and added the script to `generate-figures-tables.sh`.

jaclyn-taroni reviewed (approved):
👍🏻

sjspielman commented:
Noting I will merge this after #202 is approved.
